hawkular-inventory-impl-cassandra/src/main/java/org/hawkular/inventory/impl/cassandra/CassandraBackend.java

```diff
@@ -159,8 +170,59 @@ public CassandraBackend(RxSession session) {
     @Override
     public Iterator<Row> getTransitiveClosureOver(Row startingPoint, Relationships.Direction direction,
                                                        String... relationshipNames) {
-        //TODO implement
-        return null;
+        String startingCP = startingPoint.getString(Statements.CP);
+        if (startingCP == null) {
+            // Not a vertex
+            return Collections.<Row>emptyList().iterator();
+        }
+        Observable<Row> closureContent = getRelationshipTables(direction)
+                .flatMap(table -> {
+                    Statement q = select().all().from(table).where(in(Statements.NAME, Arrays.asList(relationshipNames)));
```

Comment:
I glossed over this when we were discussing your gist, but I don't know if this is really desirable.. Here you request all relationships with given name. Given we almost exclusively use this method to compute over `contains` relationships, this list is going to be absolutely huge (N less than the total number of entities in the whole inventory in all tenants, where N is the number of tenants). I understand the motivation of doing this with only a single query, but I am not sure if this is going to be feasible.

Maybe we could think about the "store whole subtree" idea circulated on the call to be used for this, too...